modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/cell_renderers/ArrayRenderer.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+function ArrayRenderer({value}) {
+	return (
+		<div>
+			{value.map((item, index) => (
+				<span key={item.id}>
+					{item.name}
```

Comment:
The `ArrayRenderer` should not make assumptions about the internal structure of the data being passed to it. As it's written now, this renderer works well as long as the `value` array contains items with an `id` and a `name` properties. That happens to be the case for the specific fields we are using to test this (info available in the `UserAccount` endpoint). However, it won't find what it expects if we use other arrays.

We can make different assumptions here. Ideally, the renderer should be able to work with "any" scalar value. In the beginning, it's fine if it just renders it as it comes, no need to assume anything else about the type of the value. 

To illustrate what I mean: above `map` can be made agnostic if it assumes data comes in a simpler form, e.g:

```
{value.map((item, index) => (
	<span key={index}>
		{item}
		{index < value.length - 1 ? ', ' : ''}
	</span>
))}
```
Given the varied nature of the involved data, it's safe to assume that this renderer is just able to output a comma-separated list of string values taken from the array. For this to work, some additional preparation may be needed before calling the renderer itself, because data is not necessarily coming in the above shape.

cc @markocikos @juanjofgliferay 

This is now [subject of internal discussion](https://liferay.slack.com/archives/C06UBV4E6BD/p1713192814758829)